Algorithms.Tests/Search/LinearSearcherTests.cs

```diff
@@ -60,7 +60,7 @@ public void FindIndex_ItemMissing_MinusOneReturned([Random(0, 1_000_000, 1_000)]
             var searcher = new LinearSearcher<int>();
             var random = Randomizer.CreateRandomizer();
             var arrayToSearch = Enumerable.Range(0, n).Select(x => random.Next(0, 1000)).ToArray();
-            var expectedIndex = -1;
+            const int expectedIndex = -1;
```

Comment:
I'd suggest removing this variable at all and inlining constant; `Assert.AreEqual(-1, actualIndex);`